packages/grpc-js/src/transport.ts

```diff
@@ -502,11 +502,14 @@ class Http2Transport implements Transport {
   ): Http2SubchannelCall {
     const headers = metadata.toHttp2Headers();
     headers[HTTP2_HEADER_AUTHORITY] = host;
-    headers[HTTP2_HEADER_USER_AGENT] = this.userAgent;
     headers[HTTP2_HEADER_CONTENT_TYPE] = 'application/grpc';
     headers[HTTP2_HEADER_METHOD] = 'POST';
     headers[HTTP2_HEADER_PATH] = method;
     headers[HTTP2_HEADER_TE] = 'trailers';
+    // Set default 'user-agent' header if it's not explicitly set in the metadata
+    if (!headers[HTTP2_HEADER_USER_AGENT]) {
+      headers[HTTP2_HEADER_USER_AGENT] = this.userAgent;
+    }
```

Comment:
1. It needs to be case-insensitive.
2. Need to add tests for it. 
3. Another concern I have is there are now two ways to set the user agent, each slightly different and having different priorities. Can you update the README to reflect this?